File: examples/shims/javascript/src/sample.js

const args = process.argv.slice(2);
console.log("sample: args", args);
switch (args[0]) {
  case "greet":
    greet(args.slice(1)[0]);
    break;
  default:
    console.log("Unknown command");
    process.exitCode = 1;
}

export function greet(msg) {
  console.log("sample: greet", msg);
  switch (msg) {
    case "hello":
      console.log("Hello, World!");
      break;
    case "goodbye":
      console.log("Goodbye, World!");
      break;
    default:
      console.log("sample: Unknown command");
      process.exitCode = 2;
  }
}

File: examples/shims/go/src/sample.go
package main

import (
	"fmt"
	"os"
)

func main() {
	args := os.Args[1:]
	fmt.Println("sample: args", args)

	if len(args) == 0 || args[0] != "greet" {
		fmt.Println("Unknown command")
		os.Exit(1)
	}

	var msg string
	if len(args) > 1 {
		msg = args[1]
	}
	greet(msg)
}

func greet(msg string) {
	fmt.Println("sample: greet", msg)
	switch msg {
	case "hello":
		fmt.Println("Hello, World!")
	case "goodbye":
		fmt.Println("Goodbye, World!")
	default:
		fmt.Println("sample: Unknown command")
		os.Exit(2)
	}
}
